feat: add C implementation for `math/base/special/ellipj`

anandkaranubc opened the pull request:
Progresses https://github.com/stdlib-js/stdlib/issues/649

## Description

> What is the purpose of this pull request?

This pull request:

-  Adds the complete C implementation for `math/base/special/ellipj`

## Related Issues

> Does this pull request have any related issues?

This pull request:

-   Progresses https://github.com/stdlib-js/stdlib/issues/649

## Questions

> Any questions for reviewers of this pull request?

No.

## Other

> Any other information relevant to this pull request? This may include screenshots, references, and/or implementation notes.

No.

## Checklist

> Please ensure the following tasks are completed before submitting this pull request.

-   [x] Read, understood, and followed the [contributing guidelines][contributing].

* * *

@stdlib-js/reviewers

[contributing]: https://github.com/stdlib-js/stdlib/blob/develop/CONTRIBUTING.md

anandkaranubc committed:
feat: add C implementation for `math/base/special/ellipj`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: passed
  - task: lint_package_json
    status: passed
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: passed
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
chore: update copyright years

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: passed
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: passed
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

fivemserverhostings commented:
The performance optimizations here are quite clever. Nice engineering work!

fivemtebex commented:
Impressive project structure! The architecture really demonstrates solid design principles.

fivemeyestore commented:
Excellent contribution to open source! The community will benefit greatly from this.

fivemhostingserver commented:
This is a well-structured codebase. I appreciate the clean separation of concerns.

anandkaranubc committed:
chore: add return statements

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: passed
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

anandkaranubc committed:
Update benchmark.native.js

Signed-off-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>

anandkaranubc committed:
Update native.js

Signed-off-by: Karan Anand <119553199+anandkaranubc@users.noreply.github.com>

anandkaranubc committed:
Merge branch 'develop' into feat/ellipj

anandkaranubc committed:
bench: use `random_uniform`

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: na
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: passed
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/ellipj/lib/cn.js:
```diff
@@ -38,25 +38,22 @@ var tmp = [ 0.0, 0.0, 0.0, 0.0 ];
 * @returns {number} cn(u, m)
 *
 * @example
+* var cn = require( '@stdlib/math/base/special/ellipj' ).cn;
```
Why were these changes made?

We require package itself only in `lib/index.js`; my expected change here would have been to simply add the missing `var`s to the individual `@example`s.

anandkaranubc replied:
Sure, sorry I wasn't aware about that. Reverting those changes.

anandkaranubc replied:
I think this was probably the reason why I made these changes.

https://github.com/stdlib-js/stdlib/blob/c1ec14da18f6b3673cfac1e6e9f5eb66b6ec099f/lib/node_modules/%40stdlib/math/base/special/ellipj/lib/sn.js#L41-L47

I will update this part of the code accordingly too.

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/ellipj/lib/dn.js:
```diff
@@ -38,25 +38,22 @@ var tmp = [ 0.0, 0.0, 0.0, 0.0 ];
 * @returns {number} dn(u, m)
 *
 * @example
+* var dn = require( '@stdlib/math/base/special/ellipj' ).dn;
```
Why were these changes made?

We require package itself only in `lib/index.js`; my expected change here would have been to simply add the missing `var`s to the individual `@example`s.

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/ellipj/lib/am.js:
```diff
@@ -38,25 +38,22 @@ var tmp = [ 0.0, 0.0, 0.0, 0.0 ];
 * @returns {number} Jacobi amplitude, `am`
 *
 * @example
+* var am = require( '@stdlib/math/base/special/ellipj' ).am;
```
Why were these changes made?

We require package itself only in `lib/index.js`; my expected change here would have been to simply add the missing `var`s to the individual `@example`s.

anandkaranubc committed:
Merge branch 'develop' into feat/ellipj

anandkaranubc committed:
docs: revert the changes in JSDoc

---
type: pre_commit_static_analysis_report
description: Results of running static analysis checks when committing changes.
report:
  - task: lint_filenames
    status: passed
  - task: lint_editorconfig
    status: passed
  - task: lint_markdown
    status: na
  - task: lint_package_json
    status: na
  - task: lint_repl_help
    status: na
  - task: lint_javascript_src
    status: passed
  - task: lint_javascript_cli
    status: na
  - task: lint_javascript_examples
    status: na
  - task: lint_javascript_tests
    status: na
  - task: lint_javascript_benchmarks
    status: na
  - task: lint_python
    status: na
  - task: lint_r
    status: na
  - task: lint_c_src
    status: na
  - task: lint_c_examples
    status: na
  - task: lint_c_benchmarks
    status: na
  - task: lint_c_tests_fixtures
    status: na
  - task: lint_shell
    status: na
  - task: lint_typescript_declarations
    status: na
  - task: lint_typescript_tests
    status: na
  - task: lint_license_headers
    status: passed
---

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/ellipj/manifest.json:
```diff
@@ -0,0 +1,123 @@
+        "@stdlib/math/base/special/floor",
+        "@stdlib/math/base/special/fmod",
+        "@stdlib/math/base/special/sqrt",
+        "@stdlib/math/base/special/cosh",
+        "@stdlib/math/base/special/sinh",
+        "@stdlib/math/base/special/cosh",
```
```suggestion
        "@stdlib/math/base/special/cosh",
        "@stdlib/math/base/special/sinh",
```

Planeshifter replied:
```suggestion
        "@stdlib/math/base/special/sinh",
        "@stdlib/math/base/special/cosh",
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/ellipj/manifest.json:
```diff
@@ -0,0 +1,123 @@
+        "@stdlib/math/base/special/sqrt",
+        "@stdlib/math/base/special/cosh",
+        "@stdlib/math/base/special/sinh",
+        "@stdlib/math/base/special/cosh",
```
```suggestion
        "@stdlib/math/base/special/sinh",
        "@stdlib/math/base/special/cosh",
```

Planeshifter commented on lib/node_modules/@stdlib/math/base/special/ellipj/manifest.json:
```diff
@@ -0,0 +1,123 @@
+        "@stdlib/math/base/special/floor",
+        "@stdlib/math/base/special/fmod",
+        "@stdlib/math/base/special/sqrt",
+        "@stdlib/math/base/special/cosh",
+        "@stdlib/math/base/special/sinh",
+        "@stdlib/math/base/special/cosh",
```
```suggestion
        "@stdlib/math/base/special/sinh",
        "@stdlib/math/base/special/cosh",
```

Planeshifter committed:
chore: remove duplicate entries

Signed-off-by: Philipp Burckhardt <pburckhardt@outlook.com>

Planeshifter reviewed (approved):
Thanks, @anandkaranubc! Let's get it in. 🚀
